packages/fhir-router/src/repo.ts

```diff
@@ -145,6 +145,12 @@ export interface FhirRepository {
    * @returns Promise to the array of search results.
    */
   searchResources<T extends Resource>(searchRequest: SearchRequest<T>): Promise<T[]>;
+
+  beginTransaction(): Promise<void>;
+
+  commitTransaction(): Promise<void>;
+
+  rollbackTransaction(): Promise<void>;
```

Comment:
I'd also propose for consideration something like this:
```ts
withTransaction<T>((db: PoolClient) => T): Promise<T>;
```

Reply:
Yep, that is much cleaner.  Thanks for suggestion 🙏